### DISCUSSION TITLE
Structure Predication with Atom-Level Restraint

### DISCUSSION THREAD

**Discussion by Wenzhao-protein** [chaidiscovery/chai-lab#159] [Ideas]
Working with restraints is a fascinating concept! When we use BioNMR for protein structure determination, we often have numerous distance restraints for structure calculations. However, sometimes these restraints are insufficient to calculate the entire structure from scratch. If the restraint method could support atom-level continuity, it could be highly beneficial for NMR-related applications.

**Comment by arogozhnikov** [comment]
Hi Wenzhao,

thanks for feedback! Can you give an example for restraints for the model? Is it atom-atom distances + torsion angles? Is it sparse or available for all residues? More details about your inputs would help.

**Reply by Wenzhao-protein** [reply]
Hi Alex,

Thank you so much for your fast reply. 

1. For NMR-based structure calculation, we obtain .upl and .lpl files during the spectral assignment steps. You can find the file format explanation here: https://cyana.org/wiki/Distance_restraint_file.
2. The information we obtain is atom-atom distance. An example file is included in the attached .zip file, which was sourced from https://nmrtist.org/home-blog?tag=dictionary.
3. Torsion angle restraints can also be obtained in a separate file format (https://cyana.org/wiki/Torsion_angle_restraint_file). Even if only distance restraints are supported, they are already quite useful. Of course, supporting torsion angle restraints in the future would be even better.
4. The distance restraints are sparse; each line shows the upper/lower distance limits between two specified atoms.

[restraints.zip](https://github.com/user-attachments/files/17829273/restraints.zip)

**Reply by YoavShamir5** [reply]
Hi, just to add to Wenzhao's great suggestion - atom-level distance restraints will also be super helpful for covalent "docking" (the two atoms involved in covalent attachment, one from the residue and another from the ligand, are often known by the user)
